internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubMessageDeserializer.java

```diff
@@ -63,6 +64,15 @@ public PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> deserialize(
             Schema providedProtocolSchema = AvroCompatibilityHelper.parse(new String(header.value()));
             value =
                 valueSerializer.deserialize(valueBytes, providedProtocolSchema, getEnvelope(key.getKeyHeaderByte()));
+          } catch (VeniceException e) {
+            /*
+             * If a Venice exception is caught when registering a new discovered schema, we don't need any special
+             * management here, delegate it to the follower deserialize to handle such case.
+             */
+            LOGGER.warn(
+                "Venice exception in registering new encountered schema in protocol header: ",
+                VENICE_TRANSPORT_PROTOCOL_HEADER,
+                e);
           } catch (Exception e) {
```

Comment:
These two exception cases look very similar to me.  What is their difference?

Reply:
You are right, I don't have to add the new VeniceException catch and it can still work. The only reason was to add a  bit precise warning message. So, if it doesn't give us enough benefits, we can simply remove it.

Reply:
I'm trying to even understand the difference between the error messages

Reply:
Okay, let me generalize the message and remove the new exception catch block